### PR TITLE
IGAPP-636: Subscribe to push notifications and update last usage

### PR DESCRIPTION
**Pull request by steffenkleinle**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

File: native/src/contexts/AppContextProvider.tsx [hunk 1]
```diff
@@ -28,15 +30,26 @@ const AppContextProvider = ({ children }: AppContextProviderProps): ReactElement
...
   const settingsResponse = useLoadAsync(loadSettings)
+  const allowPushNotifications = settingsResponse.data?.allowPushNotifications ?? null
+
+  useEffect(() => {
+    if (cityCode) {
+      dataContainer.storeLastUsage(cityCode).catch(reportError)
+    }
+  }, [cityCode])
+
+  useSubscribePushNotifications({ cityCode, languageCode, allowPushNotifications })
```
Comment (steffenkleinle) [inline native/src/contexts/AppContextProvider.tsx:52]: Not sure if this is the right place for this, let me know if you think it should be moved somewhere else.

**Review by sarahsporck — COMMENTED** [review]
PN didnt really work. Did you test this?

**Comment by steffenkleinle** [comment]
> PN didnt really work. Did you test this?

TBH not sure anymore, planned to at least haha, will check

**Comment by steffenkleinle** [comment]
@sarahsporck can not reproduce, works for fine for me (at least with a newly installed app).

**Comment by SteffiStuffel** [comment]
Could receive a PN, but get this error:

**Review by SteffiStuffel — APPROVED** [review]

**Comment by steffenkleinle** [comment]
> Could receive a PN, but get this error:
> 
> The 'navigation' object hasn't been initialized yet. This might happen if you don't have a navigator mounted, or if the navigator hasn't finished mounting. See https://reactnavigation.org/docs/navigating-without-navigation-prop#handling-initialization for more details. ERROR [TypeError: undefined is not a function]

This issue is already known and existed before the refactoring, would therefore ignore it for now: 
https://issues.tuerantuer.org/browse/IGAPP-1085

**Comment by steffenkleinle** [comment]
@sarahsporck seems that the app has to be reinstalled for it to work. We just have to check for production upgrades if that is the case as well which would not be good.

**Comment by sarahsporck** [comment]
Ok will try again. With newly installed app.

**Review by sarahsporck — APPROVED** [review]
Tested again. I didn't have the app in background 🤦🏼 Works fine 👍🏼

**Pull request merged**